mantle/cmd/kola/spawn.go

```diff
@@ -60,6 +64,7 @@ func init() {
 	cmdSpawn.Flags().StringVarP(&spawnUserData, "userdata", "u", "", "file containing userdata to pass to the instances")
 	cmdSpawn.Flags().BoolVarP(&spawnDetach, "detach", "t", false, "-kv --shell=false --remove=false")
 	cmdSpawn.Flags().BoolVarP(&spawnShell, "shell", "s", true, "spawn a shell in an instance before exiting")
+	cmdSpawn.Flags().BoolVarP(&spawnReconnect, "reconnect", "", false, "keep trying to reconnect to machine when disconnected")
```

Comment:
I think what'd be much more "Do what I mean" is if we made it so exiting the auto-login shell did `shutdown`, and we tell qemu to exit on *poweroff* (not reboot - I think that's possible).

Reply:
Or to say this another way - we should lifecycle bind `kola` to `qemu` and not `ssh`.

Reply:
Hmm, maybe. I do like that the current semantics actually match `cosa run`: if you exit the shell, it immediately logs you back in, but if you want to actually kill it, you can `poweroff`. (Actually, one issue right now with this code is that it still tries to reconnect if qemu exited because the VM powered off... need to investigate a follow-up fix for that).

Reply:
Though, perhaps the intersection of what we're talking about is that typing `poweroff` (and not ctrl-d/exit) becomes the canonical way to exit, and `--reconnect` is the default.

Reply:
QEMU already does this today by default, right? (Exiting when the VM is powered off).

> Why would you want to exit the shell and have it respawn?

Mostly for parity with `cosa run`, but also because if you're using `--reconnect`, presumably you're running some more involved tests, and doing Ctrl+D is kinda too easy to do by mistake and in the process nuke the VM (if e.g. you're exiting nested bash sessions or containers or something). `poweroff` OTOH leaves little room for interpretation.

Agreed re. possibly making this the default! (Though I'd want to fix that one issue mentioned above first -- I like the idea of tying it directly to the qemu process lifecycle, but we also need to think about how all this applies to non-qemu platforms too.)

Reply:
> QEMU already does this today by default, right? (Exiting when the VM is powered off).

Yes, but `kola spawn` prints lots of errors since it's tracking `ssh`.

Reply:
Though actually to make this work for `kola spawn` in the !qemu case gets hard; we'd need to poll the underlying IaaS I think.

Reply:
Yeah, I was thinking about that too. I think the core issue is that kola today does not track instance lifecycle since it (reasonably) assumes that it's in control of the resource.

So we should actually gate `--reconnect` behind checking that we're on qemu only otherwise we risk leaking VMs if targeting a cloud.

Reply:
https://github.com/coreos/coreos-assembler/pull/1217